### PR TITLE
Support multiple subscription identifiers

### PR DESCRIPTION
**Pull request by ansd**
The MQTT 5.0 spec states

> The Subscription Identifier can have the value of 1 to 268,435,455.
> It is a Protocol Error if the Subscription Identifier has a value of 0.
> Multiple Subscription Identifiers will be included if the publication
> is the result of a match to more than one subscription, in this case
> their order is not significant.

> If the Client specified a Subscription Identifier for any of the
> overlapping subscriptions the Server MUST send those Subscription
> Identifiers in the message which is published as the result of the
> subscriptions [MQTT-3.3.4-3]. If the Server sends a single copy of
> the message it MUST include in the PUBLISH packet the Subscription
> Identifiers for all matching subscriptions which have a Subscription
> Identifiers, their order is not significant [MQTT-3.3.4-4]. If the
> Server sends multiple PUBLISH packets it MUST send, in each of them,
> the Subscription Identifier of the matching subscription if it has a
> Subscription Identifier [MQTT-3.3.4-5].

This commit allows parsing multiple subscription identifiers if the server sends a single copy.

Technically, this client library only needs to **parse** multiple subscription identifiers. This is because the server can send multiple subscription identifiers to the client in the PUBLISH packet. However, the client is not allowed to send multiple subscription identifiers to the server in the SUBSCRIBE packet (and none at all in the PUBLISH packet). However, in order for the symmetric property test to succeed, this commit also adds the serialisation part.

Further, to keep this library backwards compatible, if the server sends a single subscription identifier, the client app will continue to receive an integer. The client app receives a list only if the server sends multiple subscription identifiers in the PUBLISH packet.

This commit also fixes a wrong type sepc in the README.

### CODE REVIEW

File: src/emqtt_frame.erl [hunk 1]
```diff
@@ -336,6 +336,9 @@ parse_properties(Bin, ?MQTT_PROTO_V5) ->
     <<PropsBin:Len/binary, Rest1/binary>> = Rest,
     {parse_property(PropsBin, #{}), Rest1}.
 
+parse_property(<<>>, Props = #{'Subscription-Identifier' := Ids})
+  when is_list(Ids) ->
+    maps:update('Subscription-Identifier', lists:reverse(Ids), Props);
```
Comment (qzhuyan) [inline src/emqtt_frame.erl:341]: maybe we don't need reverse here?

**Comment by qzhuyan** [comment]
could you rebase onto latest master?

**Comment by ansd** [comment]
Thanks for reviewing @qzhuyan. I rebased onto latest master.

File: src/emqtt_frame.erl (hunk 1 above)
Comment (ansd) [reply src/emqtt_frame.erl:341]: You're right, we don't need the `lists:reverse/1` since the order of subscription identifiers is not important according to the spec.
Advantages of the `lists:reverse/1` here are:
1. The client gets the same order of subscription identifiers as sent by the server. This may help understandability and debugging. But again, order doesn't matter.
2. The symmetric property test added by this PR ensures that serialising and parsing a packet results in the same packet. Since serialise happens in the correct order, for the test to succeed, parsing also needs to happen in the correct order. But I can re-write that test.

So, I'm happy to remove the `lists:reverse/1` and re-write the test. However, I don't think that the list reversal is a performance issue here. A client receiving a PUBLISH packet with many subscription identifiers is an edge case.

**Comment by qzhuyan** [comment]
thanks for the PR!
